File: 01_GOF23/DP_examples/Lua/03_Behavioral/Interpreter/TerminalExpression.lua
-- 终结符表达式 - 处理数字
require("AbstractExpression")

NumberExpr = ArithmeticExpr:new()

function NumberExpr:new(num)
    local o = ArithmeticExpr:new()
    o.num = num
    setmetatable(o, self)
    self.__index = self
    return o
end

function NumberExpr:interpret()
    return self.num
end

File: 01_GOF23/DP_examples/Rust/03_Behavioral/Interpreter/terminal_expression.rs
// 终结符表达式 - 处理数字
use crate::arithmetic_expr::ArithmeticExpr;

pub struct NumberExpr {
    num: f64,
}

impl NumberExpr {
    pub fn new(num: f64) -> Self {
        NumberExpr { num }
    }
}

impl ArithmeticExpr for NumberExpr {
    fn interpret(&self) -> f64 {
        self.num
    }
}